business/istio_config.go

```diff
@@ -697,15 +761,59 @@ func (in *IstioConfigService) DeleteIstioConfigDetail(ctx context.Context, clust
 		}
 	}
 
-	// We need to refresh the kube cache though at least until waiting for the object to be updated is implemented.
-	kubeCache.Refresh(namespace)
+	in.waitForObjectDeletion(ctx, cluster, details.Object)
 
 	// Remove validations for the object to refresh the validation cache.
 	in.kialiCache.Validations().Remove(models.IstioValidationKey{Name: name, Namespace: namespace, ObjectGVK: resourceType, Cluster: cluster})
 
 	return nil
 }
 
+func (in *IstioConfigService) waitForObjectDeletion(ctx context.Context, cluster string, obj client.Object) {
+	kubeCache, err := in.kialiCache.GetKubeCache(cluster)
+	if err != nil {
+		log.Errorf("Failed waiting for object to be deleted in cache. You may see stale data but the delete was processed correctly. Error: %s", err)
+		return
+	}
+
+	if err := kubernetes.WaitForObjectDeleteInCache(ctx, kubeCache, obj); err != nil {
+		// It won't break anything if we return the object before it is updated in the cache.
+		// We will just show stale data so just log an error here instead of failing.
+		log.Errorf("Failed waiting for object to deleted in cache. You may see stale data but the delete was processed correctly. Error: %s", err)
+		return
+	}
+}
+
+func (in *IstioConfigService) waitForCacheUpdate(ctx context.Context, cluster string, obj client.Object) {
+	kubeCache, err := in.kialiCache.GetKubeCache(cluster)
+	if err != nil {
+		log.Errorf("Failed waiting for object to update in cache. You may see stale data but the update was processed correctly. Error: %s", err)
```

Comment:
Change error message to make it distinct

```suggestion
		log.Errorf("Cannot get cache so cannot wait for object to update in cache. You may see stale data but the update was processed correctly. Error: %s", err)
```